pkg/controllers/cluster/cluster_controller.go

```diff
@@ -101,8 +93,8 @@ type Controller struct {
 	// When cluster is just created, e.g. agent bootstrap or cluster join, we give a longer grace period.
 	ClusterStartupGracePeriod time.Duration
 
-	ClusterTaintEvictionRetryFrequency time.Duration
-	ExecutionSpaceRetryFrequency       time.Duration
+	ResourceCleanupWaitingRetryFrequency      time.Duration
+	ExecutionSpaceRemoveWaitingRetryFrequency time.Duration
```

Comment:
Given the two are both for defining the interval to cleanup during cluster termination.
How about merging them into a single config?
Like
```golang
	// CleanupCheckInterval defines the fixed interval for polling resource deletion status
	// during cluster removal.
	// The fixed interval bypasses exponential backoff mechanisms to ensure the check frequency remains balanced
	// - neither too frequent to risk system overload nor too sparse to cause delays.
	CleanupCheckInterval time.Duration
```

Reply:
It sounds good!